http/src/main/scala/org/http4s/blaze/http/http2/HeaderDecoder.scala

```diff
@@ -0,0 +1,91 @@
+package org.http4s.blaze.http.http2
+
+import java.nio.ByteBuffer
+import java.nio.charset.StandardCharsets.US_ASCII
+
+import org.http4s.blaze.http.http2.Http2Exception._
+import org.http4s.blaze.http.http2.Http2Settings.DefaultSettings
+import org.http4s.blaze.util.BufferTools
+import com.twitter.hpack.{Decoder, HeaderListener}
+
+import scala.collection.immutable.VectorBuilder
+
+/** Decoder of HEADERS frame payloads into a collection of key-value
+  * pairs using a HPACK decoder.
+  *
+  * @param maxHeaderListSize maximum allowed size of the header block.
+  *
+  * "The value is based on the uncompressed size of header fields,
+  *  including the length of the name and value in octets plus an
+  *  overhead of 32 octets for each header field."
+  * https://tools.ietf.org/html/rfc7540#section-6.5.2
+  * @param maxTableSize maximum compression table to maintain
+  */
+final class HeaderDecoder(
+  maxHeaderListSize: Int,
+  discardOverflowHeaders: Boolean,
+  val maxTableSize: Int
+) {
+  require(maxTableSize >= DefaultSettings.HEADER_TABLE_SIZE)
+
+  private[this] val acc = new VectorBuilder[(String, String)]
+  private[this] var leftovers: ByteBuffer = null
+  private[this] var headerBlockSize = 0
+
+  private[this] val decoder = new Decoder(maxHeaderListSize, maxTableSize)
+  private[this] val listener = new HeaderListener {
+    override def addHeader(name: Array[Byte], value: Array[Byte], sensitive: Boolean): Unit = {
```

Comment:
What does `sensitive` mean?  Who is passing it to you?  I'm surprised this isn't a function of the name.

Reply:
Hpack allows you to flag a header as sensitive to signal that it shouldn't be cashed. Unfortunately there isn't a great api to forward that info.